### PR TITLE
[script] support cmake-build in different dir

### PR DESCRIPTION
**Pull request by bukepo**
This commit allows calling this cmake-build script from a different directory. Currently it only supports root of OpenThread source.

### CODE REVIEW

File: src/posix/platform/hdlc_interface.cpp [hunk 1]
```diff
@@ -330,6 +330,8 @@ void HdlcInterface::UpdateFdSet(fd_set &aReadFdSet, fd_set &aWriteFdSet, int &aM
     OT_UNUSED_VARIABLE(aWriteFdSet);
     OT_UNUSED_VARIABLE(aTimeout);
 
+    assert(mSockFd != -1);
```
Comment (simonlingoogle) [inline src/posix/platform/hdlc_interface.cpp:333]: seems unrelated change?

**Review by simonlingoogle — COMMENTED** [review]
We generally don't allow running `scripts/` script from any other directory. 

Why it's necessary to have an exception for `cmake-build`?

**Comment by bukepo** [comment]
> We generally don't allow running `scripts/` script from any other directory.
> 
> Why it's necessary to have an exception for `cmake-build`?

why not allow?

File: src/posix/platform/hdlc_interface.cpp (hunk 1 above)
Comment (zhanglongxia) [inline src/posix/platform/hdlc_interface.cpp:333]: Why we need to add `assert()` here? If the `mSockF` equals to `-1` the program will die when calling the `Init()`.

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]
